File: CS/DXSample/ViewModels/BarItemViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using System;

namespace DXSample.ViewModels {
    public class BarItemViewModel : StaticBarItemViewModel {

        Action<string> action;

        public BarItemViewModel(Action<string> action, string caption = null, Uri glyph = null) {
            this.Caption = caption;
            this.action = action;
            this.Glyph = glyph;
        }

        [Command]
        public void ExecuteAction(string parameter) {
            if (action != null) {
                action(parameter);
            }
        }

        public Uri Glyph {
            get { return this.GetValue<Uri>(); ; }
            set { this.SetValue(value); }
        }

    }
}

File: VB/DXSample/ViewModels/BarItemViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System

Namespace DXSample.ViewModels

    Public Class BarItemViewModel
        Inherits StaticBarItemViewModel

        Private action As Action(Of String)

        Public Sub New(ByVal action As Action(Of String), ByVal Optional caption As String = Nothing, ByVal Optional glyph As Uri = Nothing)
            Me.Caption = caption
            Me.action = action
            Me.Glyph = glyph
        End Sub

        <Command>
        Public Sub ExecuteAction(ByVal parameter As String)
            If action IsNot Nothing Then
                action(parameter)
            End If
        End Sub

        Public Property Glyph As Uri
            Get
                Return GetValue(Of Uri)()
            End Get

            Set(ByVal value As Uri)
                SetValue(value)
            End Set
        End Property
    End Class
End Namespace
